### PR TITLE
Build Travis CI

### PR DESCRIPTION
**Pull request by yil532**
- Initiate Travis CI build
- include docker
- include flake8

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+language: python
+python:
+  - "2.7"
+  - "3.6"
+services:
+  - docker
+before_install:
+  - pushd ..
+  - git clone https://github.com/IBM/MAX-Question-Answering-Web-App.git
+  - cd MAX-Question-Answering-Web-App
```
Comment (bdwyer2) [inline .travis.yml]:
```suggestion
```

Travis will take care of this for you

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  - docker build -t max-question-answering-web-app .
+  - docker run -it -d -p 5000:5000 max-question-answering-web-app
+  - popd
+  - sleep 30
+  - pip install -r requirements.txt
+install:
+  - python app.py &
+  - sleep 90
+before_script:
+  - pip install -r test-requirements.txt
```
Comment (bdwyer2) [inline .travis.yml:5]: This should be in the `install` section

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+script:
+  - flake8 -v
```
Comment (bdwyer2) [inline .travis.yml]:
```suggestion
  - flake8 . --max-line-length=127
```

We typically use flake8 with this setting for other MAX projects

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+language: python
+python:
+  - "2.7"
```
Comment (ptitzler) [inline .travis.yml]: Python 2.7 is no longer officially supported. No need to test this version.

File: .travis.yml (hunk 2 above)
Comment (ptitzler) [reply .travis.yml:5]: Let's start with basic testing first (ignoring Docker tests) and make sure that flake8 doesn't report any issues.

**Review by ptitzler — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+language: python
+python:
+  - "3.6"
+services:
+  - docker
+before_install:
+  - docker build -t max-question-answering-web-app .
+  - docker run -it -d -p 5000:5000 max-question-answering-web-app
+  - sleep 30
+  - pip install -r requirements.txt
```
Comment (bdwyer2) [inline .travis.yml]: This should probably be moved to the `install` section

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+install:
+  - sleep 90
```
Comment (bdwyer2) [inline .travis.yml]: Are you sure this is needed?

File: test-requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+flake8
+pytest
```
Comment (bdwyer2) [inline test-requirements.txt]: I don't think this is being used anywhere
```suggestion
```

**Review by bdwyer2 — DISMISSED** [review]

**Review by ptitzler — APPROVED** [review]
LGTM

**Pull request merged**